### DISCUSSION TITLE
New Checkout Feedback

### DISCUSSION THREAD

**Discussion by dennisreimann** [btcpayserver/btcpayserver#4308] [General]
We're curious to hear your thoughts about the new checkout, which we ship as an opt-in feature with v1.7 — let us know below!

**Comment by IamGretar** [comment]
Please add support for FixedFloat/Sideshift in checkout

**Reply by pavlenex** [reply]
It is added already?

**Reply by dennisreimann** [reply]
It's not available for the new checkout yet, but there's a PR already: Kukks/btcpayserver#8

This will be added soon. Meanwhile you have to use the classic version of the checkout.

**Comment by Wayne4167** [comment]
So how does this help me get my payment from Binance

**Reply by mp21-org** [reply]
We are NOT affiliated with the ongoing Bitcoin Bonus Scam.
In short: if you paid them anything, your money is lost.
They only use our free and open source software.

Please report these fraudsters to: https://complaint.ic3.gov/

**Comment by Wayne4167** [comment]
No help?

**Comment by Novosti1** [comment]
1. Please make timer "always shown". because right now it's not visible till last minutes : not good. we need to force customer to move their asses quck. so please fix it. it's was MAIN future - what all people loved, when you open invoce and see clock... very nice.

2. Please make WALLET number visible ( in small font ) right under QR code, because right now number is not visible at all, you can get it only if you click to "copy" - not good, customers is stupid, and it's can cause confusion for a lot of them.

3. Right now QR code have link what include ammount and another things. ( not good ) please make ability to make QR code - generate only WALLET NUMBER ( no ammount, no network ) and please make it possible to change CAPITAL LETTERS / lowercase letters.

All other things is perfect. 

I like new design better then old one.

THANK YOU


4. Also please return LOGO ( bitcoin / litecoin ) to the QR code. People love them.  Or make ability to put it on / off ( in control panel )

**Reply by dstrukt** [reply]
Really appreciate this feedback, and glad to hear you're enjoying the update .. will work this feedback into the next set of iterations!

**Reply by jgrodrigues** [reply]
I second the second and third points. Users a lot of the times want to pay with crypto using their CEX accounts, and CEX's do not accept these QR codes, they only accept the QR codes with the wallet address, and you have to manually input the btc amount to send. A better solution would be to have a button or a way to obtain a QR code with just the wallet address and not the whole payment information, besides having a way to easily copy or show the wallet address. Previously we would copy the wallet address and use a third party to generate a QR code containing only the wallet address and the customer would manually input the amount to send, when a customer was using a CEX.

**Reply by dennisreimann** [reply]
Addressed those point in #4489 and #4471. Thanks for the feedback!

**Comment by dergigi** [comment]
For zero-amount (any-amount aka top-up) invoices, unified QR code of v2 defaults to on-chain, but I'd much prefer it would default to Lightning (LNURL-pay).

On-chain is used even when "Default payment method on checkout" is Off-Chain. 

- Any amount, `LNURL-pay`: ❌ No QR code[^fn-bug]
- Any amount, `Off-Chain`: ❌ QR code shown, but QR code encodes an on-chain address
- Any amount, `On-Chain`: ✅ QR code shown, expected behavior

[^fn-bug]: Tracked as https://github.com/btcpayserver/btcpayserver/issues/4352

**Reply by mp21-org** [reply]
> For zero-amount (any-amount aka top-up) invoices, unified QR code of v2 defaults to on-chain, but I'd much prefer it would default to Lightning (LNURL-pay).

We discussed this also during final testing and came to the conclusion:
* LNURL has it's place for LNAddress and withdraws
* for Top-Up invoices we also want to offer a unified QR code, but with a 0 amount bolt11 invoice  ... it is on our agenda
* The consensus was to keep LNURL out of the new checkout as it offers only minimal benefits for this use case (sending a receipt URL to the user is nice, yeah)

None of it is set in stone of course and that's why we love your feedback to help us understand how you use btcpay.

**Reply by Wayne4167** [reply]
So what do I do now

**Reply by dennisreimann** [reply]
In case the new checkout doesn't work for you yet, you can always switch back to the classic version, which ist still the default.

We understand that it isn't fully done yet and will improve it based on the feedback we receive.

**Reply by Kukks** [reply]
> For zero-amount (any-amount aka top-up) invoices, unified QR code of v2 defaults to on-chain, but I'd much prefer it would default to Lightning (LNURL-pay).
> 
> On-chain is used even when "Default payment method on checkout" is Off-Chain. 
> 
> - Any amount, `LNURL-pay`: ❌ No QR code[^fn-bug]
> - Any amount, `Off-Chain`: ❌ QR code shown, but QR code encodes an on-chain address
> - Any amount, `On-Chain`: ✅ QR code shown, expected behavior
> 
> [^fn-bug]: Tracked as https://github.com/btcpayserver/btcpayserver/issues/4352

Honestly, the unified checkout is really not that great for flexibility on payment methods since it is just bip21 with a bolt11 lightning query string parameter. I very much prefer a different approach which expands the lightning address spec to handle other payment methods. See https://github.com/Kukks/btcpayserver/commit/7c03ca03938decb8ab4d15ef9a1525a372ae96ea 

```
I did this draft branch which does this in BTCPay a while back:
* Every invoice in btcpay becomes a lightning address
* Additional well-known paths for onchain, bolt11, 
* Additional well-known path for discovering all payment methods at the address 
```

Can someone test if the lightning query parameter can have an lnurl instead of bolt11 and if wallets handle that correctly?

0 amount bolt11 invoices offer inferior ux imo.

**Reply by mp21-org** [reply]
Regarding your last question: replacing the bolt11 invoice with a lnurl in a unified QR code seems to work:
![grafik](https://user-images.githubusercontent.com/1374810/204236354-7cf0a536-80aa-4633-bc31-65a3cd656f47.png)

QR content: bitcoin:bc1q8e7cm6p46qyljs6nsj2v59usq6tdzljad0agyy?amount=0.00002&LIGHTNING=lnurl1dp68gurn8ghj7cm0venx2efwwpjhg7nnvd5zuet49ap9gse0d3h82unv9acxz7f0dyh4js2sg3m4jm3kvdxx65ztxex4572vvex45msydmdj9

I've tested it with Breez and it defaults to the LNURL. (haven't tried actually paying, but I assume that works as well: it showed the correct amount)

So that would be our prefered for top-up invoices or the default to be going for?

**Comment by ghost** [comment]
> @dergigi

**Comment by utxo-one** [comment]
![image](https://user-images.githubusercontent.com/111649294/205449136-e599aeee-4a48-4fc0-bbc0-0727f19dae72.png)

Not sure if I'm doing it wrong, but the QR code is always showing lightning only, tested with Coinbase Wallet, Sparrow Wallet and Shakepay and none of them can pick up an on-chain address

**Reply by mp21-org** [reply]
only scenario where I noticed it was when the amount was to small (below bitcoind dust limit) i.e. with 200 sats I only get a lightning invoice.

Other then that: Bitcoin wallet is configured for the store?

**EDIT: Would it be possible to see how the Invoice Details in the backend look on your side?**

**Reply by utxo-one** [reply]
![image](https://user-images.githubusercontent.com/111649294/205449418-5245f8b0-3d41-4364-b9ea-aa5411581e46.png)
![image](https://user-images.githubusercontent.com/111649294/205449450-ab7a24da-b372-461f-8269-f910a1a78d69.png)
![image](https://user-images.githubusercontent.com/111649294/205449465-e5a2374f-a28f-4c18-8bbc-d69688b2482e.png)

**Reply by mp21-org** [reply]
looks like an nbxplorer problem. Is this mainnet or testnet? We had a similiar issue this week on testnet. Can you try if you got a newer version of nbxplorer available? I'll check in mattermost if I find the discussion....

**Reply by utxo-one** [reply]
This is mainnet. You think I should modify the docker-compose generated by btcpayserver to use a different version of nbx? I recently just ran the btcpay-update.sh i saw that the image was updated during the pull

**Reply by mp21-org** [reply]
I think there was something related to hot wallets (with or without the advanced RPC option) ... maybe try playing arround with that option... still trying to pull the discussion from mattermost.

**Reply by mp21-org** [reply]
nbxplorer tends to get better wih every new version, I wouldn't downgrade :-)

**Reply by mp21-org** [reply]
This is the discussion: https://chat.btcpayserver.org/btcpayserver/pl/yb7fg3u3k78u7ydr31zuw8z7nw
I think @NicolasDorier suggested removing that RPC config option as it does more harm then good.

**Comment by PeterRounce** [comment]
Checkout v2 is currently missing an NFC option to allow for card payments
Button is "Pay by NFC and LNURL-Withdraw" on Checkout v1

**Comment by boooyu44** [comment]
**Why is there no WALLET ADDRESS exposed on the checkout page?**
Customers are going to be confused. It shows complete lack of understanding how some payments are made and what customers need.
Not everyone is using mobile wallets or phones with cameras to make the crypto payment, plenty of people use windows electrum wallet and the QR scan functionality is broken there.
Not everyone also sets system-level payment handlers - so "pay in wallet" won't help them much. You're assuming things that are not really common for certain type of businesses. There are businesses where 80% of customers are desktop users 

Please expose the address and the amount for easy copy and paste on the new checkout page. 

The cryptocurrency transactions were always about an address and an amount, and now you're hiding them from end-user.
From a security perspective, it's also no good to teach users to blindly trust QR scans.

**Reply by dennisreimann** [reply]
Can you let us know the OS and browser? Also: http or https and with or without modal?

**Reply by dennisreimann** [reply]
Ok, this is most likely related to the iframe permissions that get set by the webserver. Can you check whether or not you are doing something extra there?

Also, what error does the browser console report? Something like this? `Uncaught (in promise) DOMException: The Clipboard API has been blocked because of a permissions policy applied to the current document. See https://goo.gl/EuHzyv for more details.`

**Reply by dennisreimann** [reply]
Fixed in #4453

**Reply by dennisreimann** [reply]
We addressed this in #4489.

**Comment by astupidmoose** [comment]
So I was finally able to enable this checkout on our site. Here is some feedback from me:

- Timer Missing: I understand the aesthetic reasoning behind this, however I feel like this kills any sense of urgency to get the payment paid in time. Obviously some don't care about this, so possibly having this as a configurable feature "Show timer if less than X minutes are left" would be a decent compromise? That way some of us can set this value to whatever amount of minutes our timer is defaulting to so it always shows, or we can set it to 5 minutes if we really don't think its important. 
- Recommended Fee: I really think this also needs to be displayed on the bottom near the button to pay in wallet. The reason being, we actually pushed for this feature to be included because we were getting so many people using fees that took weeks to get through when the chain was busy. Having this info visible is a "oh yea i better send a good fee". At the moment it seems hidden and will get lost. Not a concern right now with virtually everything going through at 1sat, but might be bad when it gets busier again. 
- "Plugins" aren't visible: Not a huge deal yet, as I doubt many use them and I really have no idea how you would go about incorporating these. 
- Logo/Name: The logo seems very small in its placement, and the Text of the Store seems bland. I would prefer being able to remove the text and replace it with a larger version of our logo. 
![image](https://user-images.githubusercontent.com/3916318/208259713-af28fb5a-db08-46db-928a-69ed6991c0e1.png)

**Reply by pavlenex** [reply]
@astupidmoose 

> "Show timer if less than X minutes are left" would be a decent compromise? 

This is exactly what we do already.

> Recommended Fee: 

I actually think we should remove this entirely, it's up to a wallet to properly recommend a fee, recommended fee isn't clear information for soemone: Which fee? Who charges this fee? Is merchant charging me a fee?

> "Plugins" aren't visible:

Which plugins I think both sideshift and the other one are there?

> Logo/Name: 

+1 for logo it really seems small cc @dennisreimann

**Reply by dennisreimann** [reply]
>> "Show timer if less than X minutes are left" would be a decent compromise?

> This is exactly what we do already.

I think what Mike is referring to is that this should be configurable, which is not the case currently.

> Recommended Fee:

I agree with Pavle here, yet the customer needs to clearly understand that they need to pay the shown amount and fee should suffice to get confirmed quickly. It seems like there are still too many problems with that and people end up underpaying.

> "Plugins" aren't visible:

SideShift and FixedFloat are compatible by now — please check if you need to upgrade them from the Manage Plugins page.

> logo it really seems small

I have an idea about how to make it configurable via CSS and also give the option to hide the store name if that's desired.

**Reply by astupidmoose** [reply]
> I think what Mike is referring to is that this should be configurable, which is not the case currently.

This is exactly what I mean. I'd like to be able to configure the warning to happen at the exact moment the invoice is generated. Some might prefer 50% of the way, some might be ok with 5 minutes. 

> > Recommended Fee:
> 
> I agree with Pavle here, yet the customer needs to clearly understand that they need to pay the shown amount and fee should suffice to get confirmed quickly. It seems like there are still too many problems with that and people end up underpaying.

While nice in theory, but most people just don't understand fee's and how they work in getting blocks into the chain. When they put 1/sat because its "cheaper" they don't realize this could potentially take weeks to confirm. Having this recommended fee saved this issue for sooooooooooo many of our customers as it made them think "oh what is this, my wallet mentions fees, I should make sure they are the same", and worked as a tool for us to also be like "we recommended you pay 50/sats but you only paid 2/sats. 

> > "Plugins" aren't visible:
> 
> SideShift and FixedFloat are compatible by now — please check if you need to upgrade them from the Manage Plugins page.

Thanks, updated and fixed. Didn't realize these would need independent updates.

**Comment by bigsk1** [comment]
Would be nice to have a screenshot or demo store to check it out and see how well we like it.

**Reply by dstrukt** [reply]
You can enable this checkout (V2) via the `Store Settings > Checkout Appearance` on the `Live Demo` from the main website.

We will definitely be surfacing more screenshots, etc.. soon though!

**Comment by chendo** [comment]
We're hoping to add Lightning as a payment method to our live store soon, however there are some UX issues that the new checkout could help to address.

We would like to use the unified QR code option, however there are still wallet apps out there that have issues. Would it be possible to display the unified QR code by default, but still have options to show a Bitcoin-only QR code? I suspect apps that support Lightning are more likely to support BIP21, so theoretically we don't need an option to also show a BOLT11-only QR code.

Thoughts?

**Reply by pavlenex** [reply]
> There are still wallet apps out there that have issues

Which apps have issue? The point of unified QR is to eliminate the options, if the wallets don't support it, it's best we ping them to do so, patching their poor work of not following specs, will only make the checkout look overcrouded and we'll be back to a legacy checkout in no time.

**Reply by dennisreimann** [reply]
I agreed with @pavlenex. If you find that the BIP21 option with unified QRs don't work for your customers just yet, opt-out of that setting and have them choose their prefered payment method explicitely. I'm sure that situation will vastly improve during this year and we've already seen increasing support since we made it more prominent on BTCPay Server.

**Reply by chendo** [reply]
It appears that Coinjar's app doesn't handle BTCPay generated QR codes even with just on-chain only... not sure what's going on there. Disregard my comment for the time being!

**Comment by chendo** [comment]
It would be good to have the option to display the fiat amount and currency where the BTC amount currently is, and potentially having the crypto amount underneath in smaller text. I know the fiat information is under View Details, but it would be great to be able to surface this easier for users.

**Reply by dstrukt** [reply]
This was in the original mocks, but we opted to remove .. but might be worth reconsidering.

Thank you for the feedback!

**Comment by esteban2006** [comment]
Hello there after updating to 1.7.5 I tried to create an invoice and I got this error 

```
Warning: You have payment methods configured but none of them match any of the requested payment methods or the rate is not available. See logs below:
02/06/2023 03:20:26:Info Creation of invoice starting
02/06/2023 03:20:26:Info BTC_USD: The rating rule is coingecko(BTC_USD)
02/06/2023 03:20:26:Info BTC_USD: The evaluated rating rule is 22899.324
02/06/2023 03:20:26:Info BTC (On-Chain): Payment method details creation took 0 milliseconds
02/06/2023 03:20:26:Error BTC: Payment method unavailable (Full node not available)
```

![btcUnable](https://user-images.githubusercontent.com/11742223/216876986-da2a9e41-b968-46bc-9749-c048e5762ab5.png)


Could anyone address me to an article or share steps to fix this 

regards,

**Reply by dennisreimann** [reply]
It looks like your full node isn't synched. Can you confirm that's the case and if so see the logs for details?

**Reply by esteban2006** [reply]
That is correct the node has stoped sync

**Comment by cahueya** [comment]
Playing around with v2 Checkout I found, when using the "Redirect invoice to redirect url automatically after paid"-Option, the v2 Checkout leads to a javascript-pagereload-loop which the v1 doesn't. I haven't investigated further but which variables are passed back that might be different?

**Reply by dennisreimann** [reply]
Thanks! Fixed in #4633

**Comment by status21** [comment]
Set it to off chain default, of chain below 50 and on chain above 500. Works like a charme. Very appreciated that customer don't have to choose with the unified QR. Genius!

**Comment by zachherbert** [comment]
Really liking the new checkout modal (via WooCommerce), one quick question – any easy way to hide the store name? I couldn't find any documentation. Thanks

<img width="454" alt="CleanShot 2023-09-28 at 12 15 30@2x" src="https://github.com/btcpayserver/btcpayserver/assets/14018740/cdd13dfb-e813-4c16-a68a-d1573f9f6bb9">

(specifically I want it to show just the logo, not the "Foundation" text)

**Reply by dennisreimann** [reply]
Yeah, in store Settings > General you can upload a Custom CSS File with this content: 

```css
.store-header .store-name { display: none; }
```

That does the trick.

**Reply by zachherbert** [reply]
perfect, much appreciated.

**Reply by dstrukt** [reply]
We could potentially build out more customizable granularity in the UI in the future as well, but glad to hear you're liking V2 @zachherbert!
